### PR TITLE
Download IPinfo ip location databases

### PR DESCRIPTION
**Pull request by joegallo**
Adds `ip_location` database download support for [IPinfo](https://ipinfo.io) databases.

### CODE REVIEW

**Review by masseyke — APPROVED** [review]

File: modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/EnterpriseGeoIpDownloader.java [hunk 1]
```diff
@@ -529,7 +541,101 @@ public CheckedSupplier<InputStream, IOException> download() {
...
+        public Checksum checksum() throws IOException {
+            final String checksumJsonUrl = this.url("mmdb/checksums"); // a minor abuse of the idea of a 'suffix', :shrug:
+            byte[] data = httpClient.getBytes(auth.get(), checksumJsonUrl); // this throws if the auth is bad
+            Map<String, Object> checksums;
+            try (XContentParser parser = XContentType.JSON.xContent().createParser(XContentParserConfiguration.EMPTY, data)) {
+                checksums = parser.map();
+            }
+            @SuppressWarnings("unchecked")
+            String md5 = ((Map<String, String>) checksums.get("checksums")).get("md5");
+            logger.info("checksum was [{}]", md5);
```
Comment (masseyke) [inline modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/EnterpriseGeoIpDownloader.java:620]: This doesn't need to be info

Comment (joegallo) [reply modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/EnterpriseGeoIpDownloader.java:620]: You're right, I'll fix it on the backport and put up a new PR to fix it on `main` tomorrow.

Comment (joegallo) [reply modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/EnterpriseGeoIpDownloader.java:620]: https://github.com/elastic/elasticsearch/pull/114872/files#diff-364b18938f2668f423e20a04dda43438d79542af891a4ed389cdf1077cb1180cR620 <-- fixed in the backport

Comment (joegallo) [reply modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/EnterpriseGeoIpDownloader.java:620]: As promised: https://github.com/elastic/elasticsearch/pull/114921

**Pull request merged**